biopandas/pdb/pandas_pdb.py

```diff
@@ -584,14 +577,104 @@ def to_pdb(self, path, records=None, gz=False, append_newline=True):
+
+        Returns
+        ---------
+        pandas.DataFrame : Pandas DataFrame object containing the start and end line indexes of the models.
+        """
+
+        other_records = self.df["OTHERS"]
+
+        idxs = other_records.loc[other_records["record_name"] == "MODEL"]
+        ends = other_records.loc[other_records["record_name"] == "ENDMDL"]
+        idxs.columns = ["record_name", "model_idx", "start_idx"]
+        idxs["end_idx"] = ends.line_idx.values
+        return idxs
+
+    def label_models(self):
+        """Adds a column (`"model_id"`) to the underlying DataFrames containing the model number."""
+        idxs = self.get_model_start_end()
+        # Label ATOMS
+        if "ATOM" in self.df.keys():
+            pdb_df = self.df["ATOM"]
+            idx_map = np.piecewise(
+                np.zeros(len(pdb_df)),
+                [(pdb_df.line_idx.values >= start_idx) & (pdb_df.line_idx.values <= end_idx) for start_idx, end_idx in zip(idxs.start_idx.values, idxs.end_idx.values)], idxs.model_idx)
+            self.df["ATOM"]["model_id"] = idx_map
+        # LABEL HETATMS
+        if "HETATM" in self.df.keys():
+            pdb_df = self.df["HETATM"]
+            idx_map = np.piecewise(
+                np.zeros(len(pdb_df)),
+                [(pdb_df.line_idx.values >= start_idx) & (pdb_df.line_idx.values <= end_idx) for start_idx, end_idx in zip(idxs.start_idx.values, idxs.end_idx.values)], idxs.model_idx)
+            self.df["HETATM"]["model_id"] = idx_map
+        if "ANISOU" in self.df.keys():
+            pdb_df = self.df["ANISOU"]
+            idx_map = np.piecewise(
+                np.zeros(len(pdb_df)),
+                [(pdb_df.line_idx.values >= start_idx) & (pdb_df.line_idx.values <= end_idx) for start_idx, end_idx in zip(idxs.start_idx.values, idxs.end_idx.values)], idxs.model_idx)
+            self.df["ANISOU"]["model_id"] = idx_map
+        return self
+
+    def get_model(self, model_index: int) -> PandasPdb:
+        """Returns a new PandasPDB object with the dataframes subset to the given model index.
+
+        Parameters
+        ----------
+        model_index : int
+            An integer representing the model index to subset to.
+
+        Returns
+        ---------
+        pandas_pdb.PandasPdb : A new PandasPdb object containing the structure subsetted to the given model.
+        """
+
+        df = deepcopy(self)
+        df.label_models()
+
+        if "ATOM" in df.df.keys():
+            df.df["ATOM"] = df.df["ATOM"].loc[df.df["ATOM"]["model_id"] == model_index]
+        if "HETATM" in df.df.keys():
+            df.df["HETATM"] = df.df["HETATM"].loc[df.df["HETATM"]["model_id"] == model_index]
+        if "ANISOU" in df.df.keys():
+            df.df["ANISOU"] = df.df["ANISOU"].loc[df.df["ANISOU"]["model_id"] == model_index]
+        return df
+
+    def get_models(self, model_indices: List[int]) -> PandasPdb:
```

Comment:
I thin the from __future__ import annotations is required from this type hint (i didn't want to hassle with defining a typevar - if you remove the import I think this needs to be removed too.

Reply:
I see. Let's leave it then